### PR TITLE
Improve scrolling within documents

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects #2051 

This is a tricky one, with a lot of competing things to balance. On the whole, I believe this is an improvement, but we should evaluate how it performs in production. My computer is likely faster than our users' machines, and so we might have a better experience. More testing is required.

The main change in this PR is to be more intentional about rendering the pages of the document. Because of the bug introduced on the zoom fix PR, we started rendering all the pages immediately. This could cause things to feel sluggish, and some interactions, like using the arrows or spacebar to scroll, would lock up.

We've done three things to fix this:
1. We now render at most 2 pages at a time. This helps prevent lockup.
1. We prioritize rendering pages that are in view or close to in view before rendering pages out of view.
1. We allow prerendering of future documents to happen, after the first five pages of the current document have been rendered.

I've also introduce a 147 page document, that is made up of repeated and merged redacted documents, to allow better testing.

### CODE REVIEW

File: client/app/components/Pdf.jsx [hunk 1]
```diff
@@ -174,19 +189,31 @@ export class Pdf extends React.PureComponent {
         // render on a canvas that has since been changed which
         // means we need to render it again.
         if (pdfDocument === this.state.pdfDocument && canvas === this.pageElements[index].canvas) {
+
+
```
Comment (mogrenAtWork) [inline client/app/components/Pdf.jsx]: Not sure if this extra newline is intended here

File: client/app/components/Pdf.jsx [hunk 2]
```diff
@@ -174,19 +189,31 @@ export class Pdf extends React.PureComponent {
...
           // If it is the same, then we mark this page as rendered
           this.setIsRendered(index, {
             pdfDocument,
             scale,
             ..._.pick(viewport, ['width', 'height'])
           });
 
+          // Since we don't know a page's size until we render it, we either use the
```
Comment (mogrenAtWork) [inline client/app/components/Pdf.jsx]: Nice comments!

File: client/app/components/Pdf.jsx [hunk 3]
```diff
@@ -174,19 +189,31 @@ export class Pdf extends React.PureComponent {
         // render on a canvas that has since been changed which
```
Comment (mogrenAtWork) [inline client/app/components/Pdf.jsx]: You might be able to split out lines 181 - 233 into their own function, this promise is growing to be pretty large

**Comment by mogrenAtWork** [comment]
Left a comment for 1 extra newline and a possible new function to split out, other than that LGTM.

I see you added a capybara test, but your logic in the Pdf.jsx file is getting kind of gnarly. You may want to add some unit tests around this, but I'll leave that to your discretion.

**Comment by mdbenjam** [comment]
I agree about adding more unit tests. However, there are some issues immediate issues that I think this PR might fix. It's also getting hard to unit test because the logic is so complex. I'm making a new tech improvement ticket to try to make the code simpler and to unit test it.

https://github.com/department-of-veterans-affairs/caseflow/issues/2385

**Comment by mogrenAtWork** [comment]
LGTM!

**Pull request merged**